### PR TITLE
Modified TFLite parser for the new dnn engine

### PR DESCRIPTION
**Pull request by alexlyulkov**
The new dnn graph is creating just by defining input and output names of each layer.
Some TFLite layers has fused activation, which doesn't have layer name and input and output names. Also some layers require additional preprocessing layers (e.g. NHWC -> NCHW). All these layers should be added to the graph with some unique layer and input and output names. 

I solve this problem by adding additionalPreLayer and additionalPostLayer layers.

If a layer has a fused activation, I add additionalPostLayer and change input and output names this way:
**original**: conv_relu(conv123, conv123_input, conv123_output)
**new**: conv(conv123, conv123_input, conv123_output_additional_post_layer) + relu(conv123_relu,  conv1_output_additional_post_layer, conv123_output)

If a layer has additional preprocessing layer, I change input and output names this way:
**original**: permute_reshape(reshape345, reshape345_input, reshape345_output)
**new**: permute(reshape345_permute, reshape345_input, reshape345_input_additional_pre_layer) + reshape(reshape345, reshape345_input_additional_pre_layer, reshape345_output)

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by asmorkalov** [comment]
Warning on my Ubuntu 18.04:
```
[ 95%] Building CXX object modules/dnn/CMakeFiles/opencv_dnn.dir/src/vkcom/src/buffer.cpp.o
/home/alexander/Projects/OpenCV/opencv-next2/modules/dnn/src/tflite/tflite_importer.cpp: In member function ‘void cv::dnn::dnn5_v20240521::TFLiteImporter::addLayer(cv::dnn::dnn5_v20240521::LayerParams&, const opencv_tflite::Operator&, bool, bool)’:
/home/alexander/Projects/OpenCV/opencv-next2/modules/dnn/src/tflite/tflite_importer.cpp:410:27: warning: ‘layerId’ may be used uninitialized in this function [-Wmaybe-uninitialized]
             dstNet.connect(it->second.first, it->second.second, layerId, i++);
             ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: modules/dnn/include/opencv2/dnn/dnn.hpp [hunk 1]
```diff
@@ -1105,23 +1105,29 @@ CV__DNN_INLINE_NS_BEGIN
 
     /** @brief Reads a network model stored in <a href="https://www.tensorflow.org/lite">TFLite</a> framework's format.
       * @param model  path to the .tflite file with binary flatbuffers description of the network architecture
+      * @param engine select DNN engine to be used. With auto selection the new engine is used first and falls back to classic.
+      * Please pay attention that the new DNN does not support non-CPU back-ends for now.
       * @returns Net object.
       */
-    CV_EXPORTS_W Net readNetFromTFLite(CV_WRAP_FILE_PATH const String &model);
+    CV_EXPORTS_W Net readNetFromTFLite(CV_WRAP_FILE_PATH const String &model, int engine=ENGINE_AUTO);
```
Comment (dkurt) [inline modules/dnn/include/opencv2/dnn/dnn.hpp:1112]: int or enum type?

File: modules/dnn/src/tflite/tflite_importer.cpp [hunk 2]
```diff
@@ -75,11 +79,12 @@ class TFLiteImporter {
 
     void parseFusedActivation(const Operator& op, ActivationFunctionType activ);
     void parseActivation(const Operator& op, const std::string& opcode, LayerParams& layerParams, bool isFused);
-    void addLayer(LayerParams& layerParams, const Operator& op);
-    int addPermuteLayer(const std::vector<int>& order, const std::string& permName, const std::pair<int, int>& inpId, int dtype);
+    void addLayer(LayerParams& layerParams, const Operator& op, bool additionalPreLayer = false, bool additionalPostLayer = false);
```
Comment (dkurt) [inline modules/dnn/src/tflite/tflite_importer.cpp:82]: @alexlyulkov may I ask please describe in few words what is the meaning of PreLayer?

**Comment by asmorkalov** [comment]
```
/home/ci/opencv/modules/dnn/src/tflite/tflite_importer.cpp:385:9: warning: 'layerId' may be used uninitialized in this function [-Wmaybe-uninitialized]

/home/ci/opencv/modules/dnn/src/tflite/tflite_importer.cpp:385:9: warning: 'layerId' may be used uninitialized in this function [-Wmaybe-uninitialized]

Build has warnings.
```

**Comment by asmorkalov** [comment]
@alexlyulkov The warning reported by the compiler looks like a bug. Please take a look.

File: modules/dnn/src/tflite/tflite_importer.cpp (hunk 2 above)
Comment (dkurt) [reply modules/dnn/src/tflite/tflite_importer.cpp:82]: @alexlyulkov, thanks!

With proposed approach of fused layers naming user has a model with unexpected layers names. What it we keep original fused layer name at the last unfused layer chain but rename only pre-layers?

```patch
If a layer has a fused activation, I add additionalPostLayer and change input and output names this way:
original: conv_relu(conv123, conv123_input, conv123_output)
- new: conv(conv123, conv123_input, conv123_output_additional_post_layer) + relu(conv123_relu, conv1_output_additional_post_layer, conv123_output)
+ new: conv_relu_pre(conv123, conv123_input, conv123_output_additional_post_layer) + conv_relu(conv123_relu, conv1_output_additional_post_layer, conv123_output)


If a layer has additional preprocessing layer, I change input and output names this way:
original: permute_reshape(reshape345, reshape345_input, reshape345_output)
new: permute_reshape_pre(reshape345_permute, reshape345_input, reshape345_input_additional_pre_layer) + permute_reshape(reshape345, reshape345_input_additional_pre_layer, reshape345_output)
```

So the motivation is to have matching output tensors by names between OpenCV and TFLite for origin layers.

Comment (alexlyulkov) [reply modules/dnn/src/tflite/tflite_importer.cpp:82]: I've added you naming for the new engine. Btw I've found out that the TFLite models don't have layer names and our parser just uses layer_outputs[0].name

File: modules/dnn/test/test_tflite_importer.cpp [hunk 3]
```diff
@@ -155,6 +155,9 @@ TEST_P(Test_TFLite, max_unpooling)
     net.setPreferableBackend(backend);
     net.setPreferableTarget(target);
 
+    if (net.getMainGraph())
```
[the remark this answers is not in the document]
Comment (alexlyulkov) [reply modules/dnn/test/test_tflite_importer.cpp:158]: Added

File: modules/dnn/include/opencv2/dnn/dnn.hpp (hunk 1 above)
Comment (asmorkalov) [reply modules/dnn/include/opencv2/dnn/dnn.hpp:1112]: I reverted back to int. Enum triggers issue with Java bindings generator. I know, it should be fixed, but it's less urgent.

**Review by dkurt — APPROVED** [review]
Looks good to me but let's wait for @vpisarev and @asmorkalov feedback

**Review by asmorkalov — APPROVED** [review]

**Pull request merged**